_rsrch/cidiface/README.md

```diff
@@ -129,6 +136,23 @@ if a situation is at the scale where it's become important to mind whether
 or not pointers are a performance impact, then that situation also
 is one where you have to think twice before using interfaces.
 
+### struct wrappers can be used in place of typedefs with zero overhead
+
+See `TestSizeOf`.
+
+Using the `unsafe.Sizeof` feature to inspect what the Go runtime thinks,
+we can see that `type Foo string` and `type Foo struct{x string}` consume
+precisely the same amount of memory.
+
+This is interesting because it means we can choose between either
+type definition with no significant overhead anywhere we use it:
+thus, we can choose freely between Option B and Option F based on which
+we feel is more pleasant to work with.
+
+Option F (a struct wrapper) means we can prevent casting into our Cid type.
+Option B (typedef string) can be declared a `const`.
+Are there any other concerns that would separate the two choices?
```

Comment:
I was going to speculate about if one would be easier/faster for refmt to munge than the other, because struct-containing-string seems like it going to require at least a few more reflection calls...

But on second thought that probably doesn't matter at all; CIDs are going to be using [transform funcs](https://godoc.org/github.com/polydawn/refmt/obj/atlas#UnmarshalTransformFunc) either way because the serial type is either bytes (which is a different `reflect.Kind` than string, so needs transform) or base-$n$ encoded string (which... needs to be de-base-$n$'d, so that's a transform).  So it's the same.

*Are* there any other things that would make B/F distinct?  :thinking: